server/monitor/src/main/resources/resources/overview.js

```diff
@@ -0,0 +1,293 @@
+  }).appendTo('#master');
+}
+
+/**
+ * Refresh the zookeeper table
+ */
+function refreshZKTable() {
+  var data = sessionStorage.zk === undefined ?
+      [] : JSON.parse(sessionStorage.zk);
+
+  $('#zookeeper tr td:first').hide();
+  $('#zookeeper tr:gt(2)').remove();
+
+  if (data.length === 0 || data.zkServers.length === 0) {
+    $('#zookeeper tr td:first').show();
+  } else {
+    var items = [];
+    $.each(data.zkServers, function(key, val) {
+      if (val.clients >= 0) {
+        items.push('<td class="left">' + val.server + '</td>');
+        items.push('<td class="left">' + val.mode + '</td>');
+        items.push('<td class="right">' + val.clients + '</td></tr>');
+      } else {
+        items.push('<tr><td class="left">' + val.server + '</td>');
+        items.push('<td class="left"><span class="error">Down</span></td>');
+        items.push('<td class="right"></td>');
+      }
+    });
+    $('<tr/>', {
+      html: items.join('')
+    }).appendTo('#zookeeper table');
+  }
+}
+
+/**
+ * Generates the zookeeper table
+ */
+function createZKTable() {
+  var items = [];
+  items.push('<tr><th colspan="3">Zookeeper</th></tr>');
+  items.push('<tr><th>Server</th><th>Mode</th><th>Clients</th></tr>');
+  items.push('<td class="center" colspan="3"><i>No Zookeepers</i></td>');
+  $('<table/>', {
+    html: items.join(''),
+    class: 'table table-bordered table-striped table-condensed'
+  }).appendTo('#zookeeper');
+}
+
+//// Overview plot creation
+
+/**
+ * Create the plots for the overview page
+ */
+function makePlots() {
+  var d = new Date();
+  var n = d.getTimezoneOffset() * 60000; // Converts offset to milliseconds
+  var tz = new Date().toLocaleTimeString('en-us',
+      {timeZoneName: 'short'}).split(' ')[2]; // Short version of timezone
+  var tzFormat = '%H:%M<br />' + tz;
+
+  // Create Ingest Rate plot
+  var ingestRate = [];
+  var data = sessionStorage.ingestRate === undefined ?
+      [] : JSON.parse(sessionStorage.ingestRate);
+  $.each(data, function(key, val) {
+
+    ingestRate.push([val.first - n, val.second]);
+  });
+  $.plot($('#ingest_entries'), [{ data: ingestRate,
+      lines: { show: true }, color: '#d9534f' }],
+      {yaxis: {}, xaxis: {mode: 'time', minTickSize: [1, 'minute'],
+      timeformat: tzFormat, ticks: 3}});
+
+  // Create Scan Entries plot
+  var scanEntries = {'Read': [], 'Returned': []};
+  var data = sessionStorage.scanEntries === undefined ?
+      [] : JSON.parse(sessionStorage.scanEntries);
+  $.each(data, function(key, val) {
+    $.each(val.second, function(key2, val2) {
+      scanEntries[val.first].push([val2.first - n, val2.second]);
+    });
+  });
+  $.plot($('#scan_entries'), [{ label: 'Read',
+      data: scanEntries.Read, lines: { show: true }, color: '#d9534f' },
+      { label: 'Returned', data: scanEntries.Returned, lines: { show: true },
+      color: '#337ab7' }], {yaxis: {}, xaxis: {mode: 'time',
+      minTickSize: [1, 'minute'], timeformat: tzFormat, ticks: 3}});
```

Comment:
Managed to get this pretty simplified :), now it is just a simple function call that has all of this in it (no more repeating the same code :)), I'll push once I finish another change on the HTML/JavaScript code.